src/stream/xml_writer.rs

```diff
@@ -30,8 +27,12 @@ pub struct XmlWriter<W: Write> {
     started_plist: bool,
     stack: Vec<Element>,
     expecting_key: bool,
-    // Not very nice
-    empty_namespace: Namespace,
+    pending_collection: Option<PendingCollection>,
+}
+
+enum PendingCollection {
+    Array,
+    Dictionary,
 }
 
 impl<W: Write> XmlWriter<W> {
```

Comment:
If you are changing anything else, please can you correct my type of `fearure` below!

Reply:
I'll include it as a separate commit :)